### PR TITLE
[10.0]-[MIG]-Business requirement deliverable crm from v8 to v10

### PR DESCRIPTION
**Pull request by YogeshMahera-SerpentCS**
Issue #48 

- depends on #223 #224 
- Migrated Business requirement deliverable crm from v8 to 10

@elicoidal @victormartinelicocorp

### CODE REVIEW

**Comment by elicoidal** [comment]
#224 and #223 are merged: please rebase

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal rebased

File: business_requirement_deliverable_crm/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-# © 2016 Elico Corp (https://www.elico-corp.com).
+# © 2017 Elico Corp (https://www.elico-corp.com).
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/__init__.py]: 2016-2017 everywhere

File: business_requirement_deliverable_crm/views/crm_view.xml [hunk 2]
```diff
@@ -1,50 +1,45 @@
...
+        <field name="view_mode">tree,form</field>
+        <field name="search_view_id" ref="business_requirement.view_business_requirement_search"/>
+        <field name="context">{
+            'search_default_project_id': [project_id],
+            'default_project_id': project_id,
+            'active_test': False,
+        }</field>
+        <field name="help" type="html">
+            <p>
+                Odoo's project management allows you to manage the pipeline
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/views/crm_view.xml]: Adapt (a text has been provided in business_requirement PR)

**Review by elicoidal — CHANGES_REQUESTED** [review]
Functional tests are not successful:
How to reproduce: go to CRM opp Business Analysis. Click on button "Generate quotation from BR"
Expected behavior: a quotation is created based on deliverables
Current behavior: an error says there is no BR available.

Another issue: There is no link to the BR in the project (it should be the case in the business_requirement module)

File: business_requirement_deliverable_crm/views/crm_view.xml (hunk 2 above)
Comment (YogeshMahera-SerpentCS) [reply business_requirement_deliverable_crm/views/crm_view.xml]: Adapted help text in PR #248

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS is it good for tests?

**Comment by elicoidal** [comment]
One last detail: in deliverable form view, the Currency field should be after the total revenue field.

**Comment by YogeshMahera-SerpentCS** [comment]
Please check this.

peek 2017-09-18 18-13crm

**Comment by elicoidal** [comment]
- [ ] No field to the BR in project:
screenshot from 2017-09-19 09-05-25
- [x] can you recheck demo data with Opp "Business Analysis" linked to project "Business Analysis": I cannot create the SQ
- [x] not sure this is about this environment but all projects are duplicated.
- [x] Created a new Opp "TEST" associated to project "Corporate Website" and I cannot create the quotation.

Please recheck @YogeshMahera-SerpentCS

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 

Could you please check it.

**Comment by elicoidal** [comment]
Creating quotation is fine but:
- [x] When updating the data in the BR0005. If I click in create quotation I should have a checkbox "Update quotation" like in v8. if BR changed, quotation is updated when selecting this box. Now even clicking on the button, the quotation is not updated (needs to delete manually the previous one)
- [x] Customer Reference field should contain the name of the BR

**Comment by elicoidal** [comment]
@victormartinelicocorp

**Comment by martintorresvm** [comment]
@YogeshMahera-SerpentCS Small details, here I found this on runbot:
```
2017-09-19 10:44:14,631 213 WARNING openerp_test odoo.models: crm.lead.create() includes unknown fields: ref
```
And just for curiosity about this change, may ask why?
https://github.com/OCA/business-requirement/pull/229/files#diff-ffb485409198eeea75ff84bfadc366fbL139

cc @elicoidal

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
>can you recheck demo data with Opp "Business Analysis" linked to project "Business Analysis": I cannot create the SQ

Can you please check again I have created SO from Opp Business Analysis it working fine.

> Created a new Opp "TEST" associated to project "Corporate Website" and I cannot create the quotation.

It's due to no relation(Corporate Website) with BR(master project). I have added remain relation project with  BR(master project) in demo data in PR #231

**Comment by YogeshMahera-SerpentCS** [comment]
Can you please review remain points.
@elicoidal

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS 

Looks better. One small detail left to fix:
- [x] the smart button BR in CRM opp is wrongly located (see below)

screenshot from 2017-09-21 09-28-52

- [x] Another detail is the name of the button could be shortened such as "New Quotation (Bus. Req.)" and the button placed after the New Quotation button.

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
Could you please review it.

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS One part looks very good but I was playing with the pricelist and there are couple of issues we need to fix:
* add a Pricelist field in the BR, to be updated when the customer is changed (same as in SO) with following rule: if PL estimation is not empty use Pricelist estimation ; otherwise use customer pricelist. Place the field in the "Other" tab (Visible only for estimation group)
* All deliverable and resources prices should be calculated based on this new field. Currency field in cost control too.
* When generating the SQ, the PL in the BR should be the one used for the SQ.

Let me know if unclear

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal 
Could you please check BR Cost calculations in this PR #225 
 it's related to cost that's why

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS Can you rebase and fix the Pricelist in the sales quotation (must the one from the BR)?

**Comment by YogeshMahera-SerpentCS** [comment]
@elicoidal rebased and Fixed issue:- SO pricelist comes from BR

**Comment by elicoidal** [comment]
@YogeshMahera-SerpentCS still some small details (but the main workflow is working)
- [x] Still no field to the BR in project:
screenshot from 2017-09-19 09-05-25
- [x] Demo data: Create HKD pricelist and set pricelist to BR00005
- [x] Add Pricelist menu entry to BR/Configuration menu

**Comment by YogeshMahera-SerpentCS** [comment]
>  Still no field to the BR in project: 
Regarding this point in Project module already defined BR field in Project
https://github.com/OCA/business-requirement/blob/10.0/business_requirement_deliverable_project/views/project.xml#L15

Other Points you can review it
@elicoidal

**Comment by elicoidal** [comment]
Runbot is not running: I will merge and we will test locally

**Comment by elicoidal** [comment]
Tested: LGTM

**Pull request merged**